### PR TITLE
refactor(useCustomizeColumns): Add Typescript Types

### LINKED ISSUE #5189: Add Typescript types to useCustomizeColumns

### PR DESCRIPTION
**Pull request by sangeethababu9223**
Closes #5189

Add Typescript types to useCustomizeColumns

#### What did you change?

Changed the `useCustomizeColumns.js` to `useCustomizeColumns.tsx` with typescript type support

#### How did you test and verify your work?

Storybook

### CODE REVIEW

File: packages/ibm-products/src/components/Datagrid/types/index.ts [hunk 1]
```diff
@@ -266,6 +278,10 @@ export interface DataGridState<T extends object = any>
   loadMoreThreshold?: number;
   onRowClick?: (row, event) => void;
   onSort?: boolean;
+  customizeColumnsProps?: {
```
Comment (szinta) [inline packages/ibm-products/src/components/Datagrid/types/index.ts:281]: do we need to add the below 2 properties to `customizeColumnProps` definition as we are setting those as well in the hook?
`isTearsheetOpen`,
`setIsTearsheetOpen`,

File: packages/ibm-products/src/components/Datagrid/types/index.ts [hunk 2]
```diff
@@ -266,6 +278,10 @@ export interface DataGridState<T extends object = any>
...
+    onSaveColumnPrefs?: (args) => void;
+    labels?: ColumnLabels;
+  };
```
Comment (szinta) [inline packages/ibm-products/src/components/Datagrid/types/index.ts:284]: Also looks like we need to add `CustomizeColumnsButton` prop in `DataGridState` as it is not present in type definition.

**Review by szinta — COMMENTED** [review]
Looks good to me. Just 2 observations related to 3 properties which are missing in type definitions

File: packages/ibm-products/src/components/Datagrid/types/index.ts (hunk 1 above)
Comment (sangeethababu9223) [reply packages/ibm-products/src/components/Datagrid/types/index.ts:281]: Updated the props.

File: packages/ibm-products/src/components/Datagrid/types/index.ts (hunk 2 above)
Comment (sangeethababu9223) [reply packages/ibm-products/src/components/Datagrid/types/index.ts:286]: Done

**Review by szinta — APPROVED** [review]

**Review by emyarod — DISMISSED** [review]
looks good to me!

**Review by szinta — APPROVED** [review]

**Review by amal-k-joy — APPROVED** [review]

**Pull request merged**